test/acceptance/tests.py

```diff
@@ -592,6 +623,210 @@ def test_file_upload(self):
+        self.staff_area_page = StaffAreaPage(self.browser, self.problem_loc)
+
+    def login_user(self, learner, email=LEARNER_EMAIL, password=LEARNER_PASSWORD):
+        """
+        Logs in an already existing user.
+
+        Args:
+            learner (str): the username of the user.
+            email (str): email (if not specified, LEARNER_EMAIL is used).
+            password (str): password (if not specified, LEARNER_PASSWORD is used).
+        """
+        auto_auth_page = AutoAuthPage(
+            self.browser, email=email, password=password, username=learner,
+            course_id=self.TEST_COURSE_ID, staff=True
+        )
+        auto_auth_page.visit()
+
+    def do_submission(self, email, password):
+        """
+        Creates a user and submission.
+
+        Args:
+            email (str): email for the new user
+            password (str): password for the new user
+
+        Returns:
+            str: the username of the newly created user
+        """
+        auto_auth_page = AutoAuthPage(
+            self.browser, email=email, password=password, course_id=self.TEST_COURSE_ID, staff=True
+        )
+        auto_auth_page.visit()
+        username = auto_auth_page.get_username()
+        self.submission_page.visit().submit_response(self.SUBMISSION)
+
+        return username
+
+    def do_submission_training_self_assessment(self, email, password):
+        """
+        Creates a user and then does submission, training, and self assessment.
+
+        Args:
+            email (str): email for the new user
+            password (str): password for the new user
+
+        Returns:
+            str: the username of the newly created user
+        """
+        username = self.do_submission(email, password)
+        self.do_training()
+        self.submit_self_assessment(self.SELF_ASSESSMENT)
+
+        return username
+
+    def verify_staff_area_fields(self, username, peer_assessments, submitted_assessments, self_assessment):
+        """
+        Verifies the expected entries in the staff area for peer assessments,
+        submitted assessments, and self assessment.
+
+        Args:
+            username (str): the username of the learner to check
+            peer_assessments: the expected fields in the peer assessment section
+            submitted_assessments: the expected fields in the submitted assessments section
+            self_assessment: the expected fields in the self assessment section
+        """
+        self.staff_area_page.visit()
+        self.staff_area_page.show_learner(username)
+        self.staff_area_page.expand_learner_report_sections()
+        self.assertEqual(peer_assessments, self.staff_area_page.status_text('peer__assessments'))
+        self.assertEqual(submitted_assessments, self.staff_area_page.status_text('submitted__assessments'))
+        self.assertEqual(self_assessment, self.staff_area_page.status_text('self__assessment'))
+
+    def test_staff_override_at_end(self):
```

Comment:
For the acceptance criteria, it was my understanding that we wanted a full workflow with all the individual parts (self training, peer grading, self grading, and staff override). We also wanted to have the staff override at the beginning and the end, and wished to cover all bugs that had previously been found.

Doing multiple simpler scenarios via ddt isn't feasible due to the setup of the test (using the ORA test course), and it will substantially increase bok choy test time. We do already have targeted staff override tests for self assessment only (test_staff_override in StaffAreaTest)  and peer assessment only (see PeerAssessmentTestStaffOverride).